internal/sdkprovider/service/kafka/kafka_quota.go

```diff
@@ -0,0 +1,217 @@
+		Optional: true,
+		Description: `
+Sets the maximum percentage of CPU time that a client group can use on request handler I/O and network threads per broker within a quota window.
+Exceeding this limit triggers throttling.
+The quota, expressed as a percentage, also indicates the total allowable CPU usage for the client groups sharing the quota.`,
+		ValidateFunc: validation.IntBetween(0, 100),
+	},
+}
+
+func ResourceKafkaQuota() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Creates and manages quotas for an Aiven for Apache Kafka® service user.",
+		ReadContext:   common.WithGenClient(resourceKafkaQuotaRead),
+		CreateContext: common.WithGenClient(resourceKafkaQuotaCreate),
+		UpdateContext: common.WithGenClient(resourceKafkaQuotaUpdate),
+		DeleteContext: common.WithGenClient(resourceKafkaQuotaDelete),
+		Timeouts:      schemautil.DefaultResourceTimeouts(),
+
+		Schema:        aivenKafkaQuotaSchema,
+		CustomizeDiff: validateKafkaQuotaDiff,
+	}
+}
+
+func resourceKafkaQuotaCreate(ctx context.Context, d *schema.ResourceData, client avngen.Client) error {
+	var (
+		project  = d.Get("project").(string)
+		service  = d.Get("service_name").(string)
+		user     = d.Get("user").(string)
+		clientID = d.Get("client_id").(string)
+
+		req kafka.ServiceKafkaQuotaCreateIn
+	)
+
+	if err := schemautil.ResourceDataGet(
+		d,
+		&req,
+		schemautil.RenameAlias("client_id", "client-id"),
+	); err != nil {
+		return err
+	}
+
+	if err := client.ServiceKafkaQuotaCreate(ctx, project, service, &req); err != nil {
+		return err
+	}
+
+	d.SetId(schemautil.BuildResourceID(project, service, clientID, user))
+
+	return resourceKafkaQuotaRead(ctx, d, client)
+}
+
+func resourceKafkaQuotaUpdate(ctx context.Context, d *schema.ResourceData, client avngen.Client) error {
+	var (
+		req kafka.ServiceKafkaQuotaCreateIn
+	)
+
+	project, service, _, _, err := schemautil.SplitResourceID4(d.Id())
+	if err != nil {
+		return err
+	}
+
+	if err := schemautil.ResourceDataGet(
+		d,
+		&req,
+		schemautil.RenameAlias("client_id", "client-id"),
+	); err != nil {
+		return err
+	}
+
+	if err := client.ServiceKafkaQuotaCreate(ctx, project, service, &req); err != nil {
+		return err
+	}
+
+	return resourceKafkaQuotaRead(ctx, d, client)
+}
+
+func resourceKafkaQuotaRead(ctx context.Context, d *schema.ResourceData, client avngen.Client) error {
+	project, serviceName, clientID, user, err := schemautil.SplitResourceID4(d.Id())
+	if err != nil {
+		return err
+	}
+
+	var params [][2]string
+	if user != "" {
+		params = append(params, kafka.ServiceKafkaQuotaDescribeUser(user))
+	}
+
+	if clientID != "" {
+		params = append(params, kafka.ServiceKafkaQuotaDescribeClientId(clientID))
+	}
+
+	resp, err := client.ServiceKafkaQuotaDescribe(
+		ctx,
+		project,
+		serviceName,
+		params...,
+	)
+	if err != nil {
+		return err
+	}
+
+	return schemautil.ResourceDataSet(
+		aivenKafkaQuotaSchema,
+		d,
+		resp,
+		schemautil.RenameAlias("client_id", "client-id"),
```

Comment:
nit: looks like it is used everywhere, probably it is better to store it in a map like here [mirrormaker_replication_flow.go](https://github.com/aiven/terraform-provider-aiven/blob/9d58d73419cc1551640df9b51438249471ff0bc1/internal/sdkprovider/service/kafka/mirrormaker_replication_flow.go#L22-L22)